x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_objects_service.ts

```diff
@@ -229,10 +225,10 @@ export class EncryptedSavedObjectsService {
       }
 
       try {
-        decryptedAttributes[attributeName] = await this.crypto.decrypt(
+        decryptedAttributes[attributeName] = (await this.crypto.decrypt(
           attributeValue,
           encryptionAAD
-        );
+        )) as string;
```

Comment:
nit: Any chance we can update the types for `node-crypto` to allow `decrypt` to accept a generic argument, like the old interface supported?

```ts
decrypt<T>(valueToDecrypt: string, aad?: string): Promise<T>;
```

Reply:
@legrego we actually had a _heated_ discussion about the same in the `node-crypto` repo but we decided not to do so for a reason:
The `valueToDecrypt` is not validated in any way before decryption (apart from the fact that it's a string when encrypted). It means that we have no control over the output of the decrypt method. 
i.e.: You might be expecting the decrypted value to be a string, but that doesn't stop anyone to provide a `valueToDecrypt` that gets decrypted into anything else (`string[]`, `object`, `boolean`, ...).
In my opinion, accepting a generic argument would allow (and encourage) devs to set the expected output, defeating the purpose of TS as a _best practice advisor_. With the current implementation (no generic argument), TS reminds you might get an unexpected output and you should implement type-guards. If you are sure about the exact output because you have full control over the input, you can cast it.

Reply:
Thanks for the explanation @afharo!